cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -344,24 +348,34 @@ handleSimpleNode p trace nodeTracers nc onKernel = do
           (appendName "signal-handler" trace)
           (meta, LogMessage (Text.pack "SIGHUP signal received - Performing topology configuration update"))
 
-      nt <- readTopologyFileOrError nc
-
-      let (localRoots, publicRoots) = producerAddresses nt
-
-      atomically $ do
-        writeTVar localRootsVar localRoots
-        writeTVar publicRootsVar publicRoots
-        writeTVar useLedgerVar (useLedgerAfterSlot nt)
-
-      traceNamedObject
-        (appendName "local-roots" trace)
-        (meta, LogMessage . Text.pack . show $ localRoots)
-      traceNamedObject
-        (appendName "public-roots" trace)
-        (meta, LogMessage . Text.pack . show $ publicRoots)
-      traceNamedObject
-        (appendName "use-ledger-after-slot" trace)
-        (meta, LogMessage . Text.pack . show $ useLedgerAfterSlot nt)
+      result <- try $ readTopologyFileOrError nc
+
+      case result of
+        Left (FatalError err) ->
```

Comment:
what is `FatalError`, is this some specaill exception thrown by `readTopologyFileOrError`?  Will this handle `IOException`s?

Reply:
FatalError is thrown by `readTopologyFileOrError` (https://github.com/input-output-hk/cardano-node/blob/p2p-master/cardano-node/src/Cardano/Node/Configuration/TopologyP2P.hs#L182)